homeassistant/components/wallbox/__init__.py

```diff
@@ -5,37 +5,43 @@
 import requests
 from wallbox import Wallbox
 
-from homeassistant import exceptions
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
-from .const import CONF_CONNECTIONS, CONF_ROUND, CONF_SENSOR_TYPES, CONF_STATION, DOMAIN
+from .const import (
+    CONF_CONNECTIONS,
+    CONF_DATA_KEY,
+    CONF_MAX_CHARGING_CURRENT_KEY,
+    CONF_ROUND,
+    CONF_SENSOR_TYPES,
+    CONF_STATION,
+    DOMAIN,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
-PLATFORMS = ["sensor"]
+PLATFORMS = ["sensor", "number"]
 UPDATE_INTERVAL = 30
 
 
-class WallboxHub:
+class WallboxHub(DataUpdateCoordinator):
     """Wallbox Hub class."""
 
-    def __init__(self, station, username, password, hass):
+    def __init__(
+        self, station: str, username: str, password: str, hass: HomeAssistant
+    ) -> None:
         """Initialize."""
         self._station = station
-        self._username = username
-        self._password = password
-        self._wallbox = Wallbox(self._username, self._password)
+        self._wallbox = Wallbox(username, password)
         self._hass = hass
```

Comment:
indeed, I did not know that self.hass existed. I changed it. I do not understand what you mean with the async_setup_entry comment. I don't think it would improve the code?

Reply:
You are constructing the `Wallbox` inside the init of the coordinator class, which means you need to pass authentication to it.
The  common way to do this is to construct the `Wallbox` object in `async_setup_entry` and then pass the object to the coordinator

Reply:
Done